### PR TITLE
List - add data-based typings

### PR DESCRIPTION
**Pull request by pavelgruba**

### CODE REVIEW

File: js/ui/collection/ui.collection_widget.base.d.ts [hunk 1]
```diff
@@ -23,19 +23,24 @@ import Widget, {
     WidgetOptions,
 } from '../widget/ui.widget';
 
-export interface SelectionChangedInfo<T = any> {
-    readonly addedItems: Array<T>;
-    readonly removedItems: Array<T>;
+export interface SelectionChangedInfo<TItem extends string | CollectionWidgetItem<any> | any = any> {
+    readonly addedItems: Array<TItem>;
```
Comment (IgnatovDan) [inline js/ui/collection/ui.collection_widget.base.d.ts:29]: Imho there should be 'any'.

1. Our docs declares addedItems as 'Array of any'
["addedItems	| Array of any | The **data** of the items that have been selected."](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxTabs/Configuration/#onSelectionChanged)

2. At runtime there are 'data' objects, https://js.devexpress.com/Demos/WidgetsGallery/Demo/TabPanel/Overview/jQuery/Light/:

image

3. 'CollectionWidgetItem' declares strict set of members (it's a wrapper around 'data' object):
https://github.com/DevExpress/DevExtreme/blob/21_2/js/ui/collection/ui.collection_widget.base.d.ts#L186

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/ui/collection/ui.collection_widget.base.d.ts (hunk 1 above)
Comment (IgnatovDan) [reply js/ui/collection/ui.collection_widget.base.d.ts:29]: My fault, there can 3 types of object: string, CollectionWidgetItem and data object.
The "`<TItem extends string | CollectionWidgetItem<any> | any = any>`" declaration is correct.

File: js/ui/list.d.ts [hunk 2]
```diff
@@ -54,77 +54,80 @@ export interface ScrollInfo {
...
-export type GroupRenderedEvent = EventInfo<dxList> & {
+export type GroupRenderedEvent<TItem extends string | Item<any> | any = any, TKey = any> = EventInfo<dxList<TItem, TKey>> & {
     readonly groupData?: any;
     readonly groupElement?: DxElement;
     readonly groupIndex?: number;
 };
 
 /** @public */
-export type InitializedEvent = InitializedEventInfo<dxList>;
+export type InitializedEvent<TItem extends string | Item<any> | any = any, TKey = any> = InitializedEventInfo<dxList<TItem, TKey>>;
```
Comment (IlyaKhD) [inline js/ui/list.d.ts]: We could extract local type:
```ts
type ItemLike = string | Item<any> | any;
```

```suggestion
export type InitializedEvent<TItem extends ItemLike = any, TKey = any> = InitializedEventInfo<dxList<TItem, TKey>>;
```

**Review by DokaRus — APPROVED** [review]
LGTM

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**